casanovo/denovo/evaluate.py

```diff
@@ -225,8 +225,14 @@ def aa_match_batch(
         # Split peptides into individual AAs if necessary.
         if isinstance(peptide1, str):
             peptide1 = re.split(r"(?<=.)(?=[A-Z])", peptide1)
+        elif peptide1 is None:
```

Comment:
**suggestion:** I'd move this check first, and then have the string check be part of the `elif`.